include/swift/AST/ASTContext.h

```diff
@@ -1200,10 +1200,16 @@ class ASTContext final {
 
   ModuleDecl *getModuleByIdentifier(Identifier ModuleID);
 
-  /// Looks up an already loaded module by its ABI name.
+  /// Looks up all modules whose real name or ABI name match ModuleName.
   ///
-  /// \returns The module if found, nullptr otherwise.
-  ModuleDecl *getLoadedModuleByABIName(StringRef ModuleName);
+  /// Modules that are being looked up by ABI name must be a dependency
+  /// of a module that has that same name as its real name, as there is no
+  /// efficient way to lazily load a module by ABI name.
+  llvm::SmallVector<ModuleDecl *, 1> *
+  getModulesByRealOrABIName(StringRef ModuleName);
```

Comment:
The API returning a pointer to a SmallVector is odd. From a quick look at the implementation — couldn't this be an llvm::ArrayRef<ModuleDecl *> ?